### PR TITLE
Local ClickHouse cluster configuration

### PR DESCRIPTION
**Pull request by slvrtrn**
On top of https://github.com/ClickHouse/clickhouse-js/pull/22, to keep it cleaner for the review.
Now we can run a ClickHouse cluster locally, with 2 nodes and Clickhouse Keeper (no Zookeeper)

Nginx serves as a load balancer to expose a single ClickHouse endpoint like it is done in the Cloud.

### CODE REVIEW

File: docker-compose.cluster.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        soft: 262144
+        hard: 262144
+    hostname: clickhouse1
+    container_name: clickhouse-js-clickhouse-server-node-1
+    ports:
+      - '8124:8123'
+      - '9000:9000'
+      - '9181:9181'
+    volumes:
+      - './.docker/clickhouse/cluster/config1.xml:/etc/clickhouse-server/config.xml'
```
Comment (mshustov) [inline docker-compose.cluster.yml]: Optional: what if we give them more descriptive names? `config1.xml` --> `server_1_config.xml`?

File: __tests__/integration/insert.test.ts [hunk 2]
```diff
@@ -2,20 +2,43 @@ import { expect } from 'chai';
...
+import { TestEnv } from '../utils/client';
 
 describe('insert', () => {
   let client: ClickHouseClient;
   let tableName: string;
   beforeEach(async () => {
-    client = await createTestClient();
+    client = await createTestClient({
+      clickhouse_settings: {
+        insert_quorum: 2,
```
Comment (mshustov) [inline __tests__/integration/insert.test.ts]: Is it still necessary for the tests?

File: src/client.ts [hunk 3]
```diff
@@ -140,13 +141,16 @@ export class ClickHouseClient {
...
+    const format = params.format ?? 'JSON';
+    const query = formatCommandQuery(params.query, format);
 
-    await this.connection.command({
+    const stream = await this.connection.command({
       query,
       ...this.getBaseParams(params),
     });
+
+    return new Rows(stream, format);
```
Comment (mshustov) [inline src/client.ts:153]: Thinking loudly: we need to rename the result of the operation. not a blocker for the current PR

File: __tests__/utils/client.ts [hunk 4]
```diff
@@ -57,6 +68,19 @@ function getFromEnv(key: string): string {
...
 
-function isClickHouseCloudEnabled() {
-  return process.env['CLICKHOUSE_CLOUD_ENABLED'] === 'true';
+export function getClickHouseTestEnvironment(): TestEnv {
+  let env;
+  switch (process.env['CLICKHOUSE_TEST_ENVIRONMENT']) {
+    case 'CLOUD':
+      env = TestEnv.Cloud;
+      break;
+    case 'LOCAL_CLUSTER':
```
Comment (mshustov) [inline __tests__/utils/client.ts:77]: Where do we set it? Shouldn't we add a command to `package.json` to run the tests against a local cluster? Can be done in follow-up, though.

File: __tests__/integration/insert.test.ts (hunk 2 above)
Comment (slvrtrn) [reply __tests__/integration/insert.test.ts]: removed

File: __tests__/utils/client.ts (hunk 4 above)
Comment (slvrtrn) [reply __tests__/utils/client.ts:77]: currently in GHA only: https://github.com/ClickHouse/clickhouse-js/blob/97db905e487be69ecc472d91a4b2230fe50654cb/.github/workflows/run-tests.yml#L55

but I agree

**Pull request merged**